stack/core/src/main/java/org/apache/usergrid/corepersistence/GuiceFactory.java

```diff
@@ -0,0 +1,159 @@
+
+    private Injector injector;
+
+
+
+    public GuiceFactory( final CassandraService cass, final Properties systemProperties ) {
+        this.cass = cass;
+        this.systemProperties = systemProperties;
+    }
+
+
+    @Override
+    public Injector getObject() throws Exception {
+
+        if ( this.injector != null ) {
+            return injector;
+        }
+
+
+        try {
+
+            cass.init();
+            logger.info( "Loading Core Persistence properties" );
+
+            String hostsString = "";
+            CassandraHost[] hosts = cass.getCassandraHostConfigurator().buildCassandraHosts();
+            if ( hosts.length == 0 ) {
+                throw new RuntimeException( "Fatal error: no Cassandra hosts configured" );
+            }
+            String sep = "";
+            for ( CassandraHost host : hosts ) {
+                if ( StringUtils.isEmpty( host.getHost() ) ) {
+                    throw new RuntimeException( "Fatal error: Cassandra hostname cannot be empty" );
+                }
+                hostsString = hostsString + sep + host.getHost();
+                sep = ",";
+            }
+
+            logger.info( "hostsString: " + hostsString );
+
+            Properties cpProps = new Properties();
+
+            // Some Usergrid properties must be mapped to Core Persistence properties
+            cpProps.put( "cassandra.hosts", hostsString );
+            cpProps.put( "cassandra.port", hosts[0].getPort() );
+            cpProps.put( "cassandra.cluster_name", cass.getProperties().get( "cassandra.cluster" ) );
+
+            String cassRemoteString = ( String ) cass.getProperties().get( "cassandra.use_remote" );
+            if ( cassRemoteString != null && cassRemoteString.equals( "false" ) ) {
+                cpProps.put( "cassandra.embedded", "true" );
+            }
+            else {
+                cpProps.put( "cassandra.embedded", "false" );
+            }
+
+            cpProps.put( "collections.keyspace.strategy.class",
+                cass.getProperties().get( "cassandra.keyspace.strategy" ) );
+
+            cpProps.put( "collections.keyspace.strategy.options",
+                cass.getProperties().get( "cassandra.keyspace.replication" ) );
+
+
+            logger.debug( "Set Cassandra properties for Core Persistence: " + cpProps.toString() );
+
+            // Make all Usergrid properties into Core Persistence config
+            cpProps.putAll( cass.getProperties() );
+            cpProps.putAll( systemProperties );
+            //logger.debug("All properties fed to Core Persistence: " + cpProps.toString() );
+
+
+
+            ConfigurationManager.loadProperties( cpProps );
+        }
+        catch ( Exception e ) {
+            throw new RuntimeException( "Fatal error loading configuration.", e );
+        }
+
+        //this is seriously fugly, and needs removed
+        injector = Guice.createInjector( new GuiceModule( applicationContext ) );
```

Comment:
Could we separate this out? It seems like a method called getObject shouldn't do configuration loading either.